src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx

```diff
@@ -409,7 +410,7 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
             onSubmit={this.handleSubmit}
             onSubmitSuccess={onSubmitSuccess}
             onCancel={this.handleCancel}
-            onFieldChange={this.handleFieldChange}
+            onFieldChange={debounce(this.handleFieldChange,200)}
```

Comment:
The debounce is a good idea, but when I was poking it locally it felt like some of my changes didn't reflect in the chart until I refocused the query input and changed the text again.